tools/tf2openapi/tfmodel/generate.go

```diff
@@ -0,0 +1,122 @@
+package tfmodel
+
+import (
+	"github.com/golang/protobuf/proto"
+	pb "github.com/tensorflow/tensorflow/tensorflow/go/core/protobuf"
+	"log"
+	"strings"
+)
+
+/**
+Raises errors when model is missing fields that would pose an issue for Schema generation
+ */
+
+const servingMetaGraphTag string = "serve"
+const b64Indicator string = "_bytes"
+const predictReqSigDefMethod string = "tensorflow/serving/predict"
+
+func UnmarshalSavedModelPb(in []byte) pb.SavedModel {
+	model := &pb.SavedModel{}
+	if err := proto.Unmarshal(in, model); err != nil {
+		log.Fatalln("SavedModel not in expected format. May be corrupted.")
+	}
+	return *model
+}
+
+func tagInList(tag string, tagList []string) (bool) {
+	for _, t := range tagList {
+		if t == tag {
+			return true
+		}
+	}
+	return false
+}
+
+func GenerateTFModel(model pb.SavedModel) TFSavedModel {
+	tfSavedModel := TFSavedModel{}
+	tfMetaGraphs := []TFMetaGraph{}
```

Comment:
I think we can improve this pattern. 

This function takes on a few too many concerns and has multiple layers of nesting. Think of each for loop as a logical construct (i.e a data transformation), and encapsulate that logic into a function.

i.e.
```
for _ foo := range foos {
  bars.append(fooToBar(foo));
}

func fooToBar(foo) bar {
}
```